ethcore/account-state/src/state.rs

```diff
@@ -759,20 +759,21 @@ impl<B: Backend> State<B> {
 	}
 
 	/// Remove any touched empty or dust accounts.
-	pub fn kill_garbage(&mut self, touched: &HashSet<Address>, remove_empty_touched: bool, min_balance: &Option<U256>, kill_contracts: bool) -> TrieResult<()> {
-		let to_kill: HashSet<_> = {
-			self.cache.borrow().iter().filter_map(|(address, ref entry)|
-				if touched.contains(address) && // Check all touched accounts
-					((remove_empty_touched && entry.exists_and_is_null()) // Remove all empty touched accounts.
+	pub fn kill_garbage(&mut self, touched: &HashSet<Address>, min_balance: &Option<U256>, kill_contracts: bool) -> TrieResult<()> {
+		let to_kill: HashSet<_> =
+			touched.iter().filter_map(|address| { // Check all touched accounts
+				if let Some(entry) = self.cache.borrow().get(address) {
+					if entry.exists_and_is_null() // Remove all empty touched accounts.
```

Comment:
if you replace `if let Some(entry) = self.cache.borrow().get(address)` with `and_then`. Then you can eliminate one of the ugly `else` branches.